packages/web/src/modules/zoom/components/controls/host-control.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback } from 'react';
+
+import { ui } from '@application';
+import { Components } from '@types';
+
+import { ZoomContextType, useZoomContext } from '../../context';
+import { authorizeZoom } from '../../services';
+
+const { Button, Form } = ui;
+
+interface ZoomHostData {
+  meetingNumber: string;
+  password: string;
+}
+
+const ZoomHostControl: Components['ZoomHostControl'] = () => {
+  const zoomContext: ZoomContextType = useZoomContext();
+
+  const onSubmit = useCallback(
+    ({ meetingNumber, password }: ZoomHostData) => {
+      zoomContext.updateContext((prevState: ZoomContextType) => ({
+        ...prevState,
+        meetingNumber: meetingNumber.replaceAll(' ', ''),
+        password,
+      }));
+    },
+    [zoomContext],
+  );
+
+  if (zoomContext.password && zoomContext.meetingNumber) {
```

Comment:
Only like this: https://github.com/MrBr/chess-tent/pull/7/commits/427de9942992d07684cf1fa90f75a1d3fe1a53a0

Works, but I've got a slight delay now when the zoom call appears until the Form disappears.

Reply:
Is there a transitioning state like `connecting` or something like that?

Reply:
Negative

Reply:
In that case, you set the initial state to `connecting` when user starts joining the room.

I'd also change state a bit then. Instead of having `isOnCall` I'd save `connectionStatus`. Then you can check status like

```
context.connectionStatus === ZoomConnectionStatus.Connecting
```